### ISSUE TITLE
IstioOperator reverts any change on port number or host

### ISSUE THREAD

**Issue by julienperignon** [istio/istio#24007]
**Bug description**
With the IstioOperator, when i change the gateway port number to 443 (instead of 80) and/or host to 'gw.domain.com' after some time the IstioOperator reverts back the port number to 80 and host to '*'.
I'm running behind an AWS LoadBalancer with the SSL certificate installed 
I would expect this to be customizable from the IstioOperator spec but couldn't find anything in the documentation.

**Expected behavior**
On the gateway : port number value to remain443 , and host value to remain 'gw.domain.com'
**Steps to reproduce the bug**

```
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  namespace: istio-system
  name: istiocontrolplane
spec:
  profile: default
  addonComponents:
    prometheusOperator:
      enabled: true
    prometheus:
      enabled: false
    kiali:
      enabled: false
  components:
     ingressGateways:
     - enabled: true
       name: gw
       namespace: default
       k8s:
         serviceAnnotations:
           external-dns.alpha.kubernetes.io/hostname: gw.domain.com
           service.beta.kubernetes.io/aws-load-balancer-ssl-cert: XXX
---
apiVersion: networking.istio.io/v1beta1
kind: Gateway
metadata:
  annotations:
    install.operator.istio.io/chart-owner: IngressGateways
  labels:
    install.operator.istio.io/owner-group: install.istio.io
    install.operator.istio.io/owner-kind: IstioOperator
    install.operator.istio.io/owner-name: istiocontrolplane
    release: istio
  name: gw
spec:
  selector:
    app: istio-ingressgateway
    istio: ingressgateway
  servers:
  - hosts:
    - '*'
    port:
      name: http
      number: 443
      protocol: HTTP
```

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
client version: 1.5.2
control plane version: 1.5.2
data plane version: 1.5.2 (12 proxies)
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2", GitCommit:"52c56ce7a8272c798dbc29846288d7cd9fbae032", GitTreeState:"clean", BuildDate:"2020-04-17T08:59:10Z", GoVersion:"go1.14.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.11", GitCommit:"d94a81c724ea8e1ccc9002d89b7fe81d58f89ede", GitTreeState:"clean", BuildDate:"2020-03-12T21:00:06Z", GoVersion:"go1.12.17", Compiler:"gc", Platform:"linux/amd64"}

**How was Istio installed?**
Via the Operator

**Environment where bug was observed (cloud vendor, OS, etc)**
AWS (via kops)

**Comment by howardjohn** [comment]
I think the problem is we are automatically creating the Gateway object you have there as well, so it overrides it. This was generally decided to be a bad idea, so we stopped doing that in 1.6: https://github.com/istio/istio/issues/22095

**Comment by julienperignon** [comment]
@howardjohn thanks for your reply. Are you saying there is no way to get this working in 1.5? Any rough idea of when a release with those suggested changes will happen?

**Comment by howardjohn** [comment]
Yes - this is not possible to work around in 1.5 as far as  iknow, but fixed in 1.6 which is shipped
